diesel/src/expression/sql_literal.rs

```diff
@@ -5,48 +5,128 @@ use expression::*;
 use query_builder::*;
 use query_dsl::RunQueryDsl;
 use result::QueryResult;
+use serialize::ToSql;
+use sql_types::HasSqlType;
 
 #[derive(Debug, Clone)]
 /// Returned by the [`sql()`] function.
 ///
 /// [`sql()`]: ../dsl/fn.sql.html
-pub struct SqlLiteral<ST> {
+pub struct SqlLiteral<ST, T = ()> {
     sql: String,
+    inner: T,
     _marker: PhantomData<ST>,
 }
 
-impl<ST> SqlLiteral<ST> {
+impl<ST, T> SqlLiteral<ST, T> {
     #[doc(hidden)]
-    pub fn new(sql: String) -> Self {
+    pub fn new(sql: String, inner: T) -> Self {
         SqlLiteral {
             sql: sql,
+            inner: inner,
             _marker: PhantomData,
         }
     }
+
+    /// Bind a value for use with this SQL query.
+    ///
+    /// # Safety
+    ///
+    /// This function should be used with care, as Diesel cannot validate that
+    /// the value is of the right type nor can it validate that you have passed
+    /// the correct number of parameters.
+    ///
+    /// # Examples
+    ///
+    /// ```rust
+    /// # #[macro_use] extern crate diesel;
+    /// # include!("../doctest_setup.rs");
+    /// #
+    /// # table! {
+    /// #    users {
+    /// #        id -> Integer,
+    /// #        name -> VarChar,
+    /// #    }
+    /// # }
+    /// #
+    /// # fn main() {
+    /// #     use self::users::dsl::*;
+    /// #     use diesel::dsl::sql;
+    /// #     use diesel::sql_types::{Integer, Text};
+    /// #     let connection = establish_connection();
+    /// let seans_id = sql::<Integer>("SELECT id FROM users WHERE name = ")
```

Comment:
Maybe we should use some other example here, because `sql` is meant to implement small missing query fragments not whole queries. So maybe just show something where a custom filter is set?

Reply:
This is a good point. I'll update this test case.